src/Protocols.Abstractions/Features/IConnectionTransportFeature.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO.Pipelines;
+using System.Text;
+
+namespace Microsoft.AspNetCore.Protocols.Features
+{
+    public interface IConnectionTransportFeature
```

Comment:
This interface needs some love. We need to represent 2 points of view:

1. Transport -> Application
2. Application -> Transport

The `ConnectionContext` is the thing the middleware sees, it's the application facing feature. This feature should be the Transport -> Application view. It should be what the application needs from the transport.

There should be another feature that gets added when OnConnection is called that does Transport -> Application. Maybe `IApplicationConnectionFeature`.